trace_decoder/tests/simulate-execution.rs

```diff
@@ -0,0 +1,34 @@
+//! Check that the [`evm_arithmetization::GenerationInputs`] produced by
+//! [`trace_decoder`] are consistent between each other, and with the block
+//! header obtained over RPC.
+
+mod common;
+
+use anyhow::Context as _;
+use common::{cases, Case};
+use libtest_mimic::{Arguments, Trial};
+use plonky2::field::goldilocks_field::GoldilocksField;
+
+fn main() -> anyhow::Result<()> {
+    let mut trials = vec![];
+    for Case {
+        name,
+        header: _,
+        trace,
+        other,
+    } in cases()?
+    {
+        let gis = trace_decoder::entrypoint(trace, other, 3, false)
+            .context(format!("error in `trace_decoder` for {name}"))?;
```

Comment:
Unrelated to this PR, but this makes me realize we _may_ want to test consistency of the decoder for both unitary batches (i.e. size 1) and larger ones (here 3).

Reply:
Gonna try and sneak this change in with a9602c87af62e989579a745d2a6bd78fc31799a9